libsolidity/ast/Types.cpp

```diff
@@ -3139,13 +3139,13 @@ string FunctionType::externalSignature() const
 
 	// "inLibrary" is only relevant if this is not an event.
 	bool const inLibrary = kind() != Kind::Event && dynamic_cast<ContractDefinition const&>(*m_declaration->scope()).isLibrary();
-	FunctionTypePointer external = interfaceFunctionType();
-	solAssert(!!external, "External function type requested.");
-	auto parameterTypes = external->parameterTypes();
-	auto typeStrings = parameterTypes | boost::adaptors::transformed([&](TypePointer _t) -> string
+	auto typeStrings = m_parameterTypes | boost::adaptors::transformed([&](TypePointer _t) -> string
 	{
-		solAssert(_t, "Parameter should have external type.");
-		string typeName = _t->signatureInExternalFunction(inLibrary);
+		auto const& _it = _t->interfaceType(inLibrary);
+		solAssert(_it.get(), "Parameter should have external type.");
```

Comment:
```suggestion
		solAssert(!!it, "Parameter should have external type.");
```
(not sure about the ``!!`` myself, but it's used a lot...)

Reply:
that doesn't actually compile. I'd need to use `!!it.get()`